components/hidden_data_box/commons/hidden_data_box.lua

```diff
@@ -0,0 +1,70 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:HiddenDataBox
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Variables = require('Module:Variables')
+local String = require('Module:StringUtils')
+
+local HDB = {}
+
+function HDB.run(args)
+	local tournament = args.tournament or mw.title.getCurrentTitle().prefixedText
+	local parent = tournament:gsub('/.-$', ''):gsub(' ', '_')
+
+	local queryResult = mw.ext.LiquipediaDB.lpdb('tournament', {
+		conditions = '[[pagename::' .. parent .. ']]',
+		limit = 1,
+		order = 'pagename asc',
+	})
+	queryResult = queryResult[1] or {}
+
+	local startDate = HDB:cleanDate(args.sdate, args.date)
+	local endDate = HDB:cleanDate(args.edate, args.date)
+	HDB:checkAndAssign('tournament_startdate', startDate, queryResult.startdate)
+	HDB:checkAndAssign('tournament_enddate', endDate, queryResult.enddate)
+
+	HDB:checkAndAssign('tournament_name', args.name, queryResult.name)
+	HDB:checkAndAssign('tournament_series', args.series, queryResult.series)
+	HDB:checkAndAssign('tournament_shortname', args.shortname, queryResult.shortname)
+	HDB:checkAndAssign('tournament_tickername', args.tickername, queryResult.tickername)
+	HDB:checkAndAssign('tournament_icon', args.icon, queryResult.icon)
+	HDB:checkAndAssign('tournament_icondark', args.icondark or args.icondarkmode, queryResult.icondark)
+	HDB:checkAndAssign('tournament_liquipediatier', args.liquipediatier, queryResult.liquipediatier)
+	HDB:checkAndAssign('tournament_liquipediatiertype', args.liquipediatiertype, queryResult.liquipediatiertype)
+	HDB:checkAndAssign('tournament_type', args.type, queryResult.type)
+	HDB:checkAndAssign('tournament_status', args.status, queryResult.status)
+	HDB:checkAndAssign('tournament_game', args.game, queryResult.game)
+	HDB:checkAndAssign('tournament_parent', args.parent, parent)
+	HDB:checkAndAssign('tournament_parentname', args.parentname, queryResult.name)
+
+	HDB:addCustomVariables(args, queryResult)
+end
+
+function HDB:cleanDate(primaryDate, secondaryDate)
```

Comment:
Seems like we can adapt https://liquipedia.net/commons/Module:ReferenceCleaner and use that, no?

Reply:
you mean like it is now, or do you mean creating a function for it in the `ReferenceCleaner` Module?

Reply:
Expanding the current function in ReferenceCleaner to support your use case

Reply:
could also restructure it to this
```lua
function HiddenDataBox:cleanDate(primaryDate, secondaryDate)
	local date = ReferenceCleaner.clean(primaryDate)
	if date == '' then
		date = ReferenceCleaner.clean(secondaryDate)
		if date == '' then
			return nil
		end
	end

	return date
end
```
____________
or add the following function in the ReferenceCleaner
```lua
function p.cleanMulti(...)
	--pack varargs
	local dates = { n = select("#", ...), ... }

	for i = 1, varargs.n do
		local date = p.clean(dates[i])
		if date ~= '' then
			return date
		end
	end
end
```

Reply:
I'd say go with option 1 :)